vscode-antimony/src/server/stibium/stibium/types.py

```diff
@@ -204,10 +204,16 @@ def __init__(self, range, val):
         self.message = ("Parameter '{}' defined but not used").format(val)
 
 class UninitCompt(Issue):
-    def __init__(self, range, val): 
+    def __init__(self, range, val):
         super().__init__(range, IssueSeverity.Warning)
         self.val = val
         self.message = ("Compartment '{}' has not been initialized, using default value").format(val)
+        
+class RateRuleNotInReaction(Issue):
+     def __init__(self, range, val):
+        super().__init__(range, IssueSeverity.Warning)

```

Comment:
I think this should be an error, assuming you've caught it properly.  If a species both appears in a reaction and has a rate rule, it *must* be a boundary species.  (In Antimony, either have a '$' by it somewhere, or be flagged as 'const' or 'ext'.)